lib/tasks/search.rake

```diff
@@ -0,0 +1,245 @@
+require 'rake'
+require 'digest/sha1'
+require 'ansi'
+
+namespace :search do
+
+  desc 're-index all records, in parallel'
+  task :index do
+    nprocs     = ENV['NPROCS'] || 1
+    batch_size = ENV['BATCH']  || 100
+    max        = ENV['MAX']    || nil
+    klass      = eval( ENV['CLASS']  || 'Manifest' )
+    nprocs     = nprocs.to_i
+    batch_size = batch_size.to_i
+
+    create_mindex({
+      :klass => klass, 
+      :idx_name => klass.index_name,
+      :nprocs => nprocs.to_i, 
+      :batch_size => batch_size.to_i, 
+      :max => max, 
+      :force => ENV['FORCE']
+    })
+  end
+
+  def create_mindex(opts)
```

Comment:
this method is doing a lot -- what do you think about putting this code / logic into a testable class? Rake tasks are a pain to test, so don't usually test them, but I do usually like to have the invoke well-tested classes.